src/vl-map-search.js

```diff
@@ -123,50 +125,117 @@ export class VlMapSearch extends vlElement(HTMLElement) {
-                };
-              });
-              this._selectElement.choices = resultaten;
-            }
-          });
+          const lambertCoordinaat = LambertCoordinaat.of(event.detail.value);
+
+          if (lambertCoordinaat === undefined) {
+            this._searchChoicesByValue(event.detail.value);
+          } else {
+            this._searchChoicesByLambertCoördinaat(lambertCoordinaat);
+          }
         }
       });
     }
   }
 
+  _searchChoicesByValue(searchValue) {
+    fetch(this.searchUrl + searchValue).then((response) => {
+      return response.json();
+    }).then((data) => {
+      if (data && data.SuggestionResult) {
+        const resultaten = data.SuggestionResult.map((resultaat) => {
+          return {
+            value: resultaat,
+            label: resultaat,
+          };
+        });
+        this._selectElement.choices = resultaten;
+      }
+    });
+  }
+
+  _searchChoicesByLambertCoördinaat(lambertCoordinaat) {
+    fetch(this._findLocationsByLambertCoordinaatUrl(lambertCoordinaat)).then((response) => {
+      return response.json();
+    }).then((data) => {
+      const choices = [{
+        value: lambertCoordinaat,
+        label: 'Lambert-coördinaat: ' + lambertCoordinaat.toString(),
+      }];
+
+      if (data && data.LocationResult) {
+        data.LocationResult.map((locationResult) => {
+          return {
+            value: locationResult,
+            label: locationResult.FormattedAddress,
+          };
+        }).forEach((choice) => {
+          choices.push(choice);
+        });
+      }
+      this._selectElement.choices = choices;
+    });
+  }
+
   _addChoiceEventListener() {
     if (!this.__choiceEventListenerRegistered) {
       this.__choiceEventListenerRegistered = true;
       this.__choiceEventListener = this._selectElement.addEventListener('choice', (event) => {
         if (event && event.detail && event.detail.choice) {
-          fetch(this.locationUrl + event.detail.choice.value).then((response) => {
-            return response.json();
-          }).then((data) => {
-            if (data && data.LocationResult) {
-              if (this._onSelect) {
-                this._onSelect(data);
-              } else if (this._map) {
-                this._map.zoomTo([
-                  data.LocationResult[0].BoundingBox.LowerLeft.X_Lambert72,
-                  data.LocationResult[0].BoundingBox.LowerLeft.Y_Lambert72,
-                  data.LocationResult[0].BoundingBox.UpperRight.X_Lambert72,
-                  data.LocationResult[0].BoundingBox.UpperRight.Y_Lambert72,
-                ]);
-              }
-            }
-          });
+          const value = event.detail.choice.value;
+
+          if (this._isLambertCoördinaat(value)) {
+            this._zoomToLambertCoördinaat(value);
+          } else if (value instanceof Object) {
+            this._zoomToLocation(value);
+          } else {
+            this._searchAndZoomToLocation(value);
+          }
```

Comment:
Ik zou deze if else boom vervangen door een `_getBoundingBox(value)` en daar een `zoomTo` van nemen. In de `_getBoundingBox` functie kan dan wel de if else boom of een switch case zitten.